website/addons/wiki/views.py

```diff
@@ -245,14 +219,45 @@ def project_wiki_delete(auth, wname, **kwargs):
 
 
 @must_be_valid_project  # returns project
-@must_have_permission('write')  # returns user, project
-@must_not_be_registration
+@must_be_contributor_or_public
 @must_have_addon('wiki', 'node')
-def project_wiki_edit(auth, wname, **kwargs):
+def project_wiki_view(auth, wname, path=None, **kwargs):
     node = kwargs['node'] or kwargs['project']
-    wiki_name = wname.strip()
+    anonymous = has_anonymous_link(node, auth)
+    wiki_name = (wname or '').strip()
     wiki_key = to_mongo_key(wiki_name)
     wiki_page = node.get_wiki_page(wiki_name)
+    toc = _serialize_wiki_toc(node, auth=auth)
+    can_edit = node.has_permission(auth.user, 'write') and not node.is_registration
+    versions = _get_wiki_versions(node, wiki_name, anonymous=anonymous)
+
+    # Parse URL to determine default panel display
+    view_settings = {}
+    param_list = path.lower().split('/') if path else []
+    for i, param in enumerate(param_list):
+        configurable = param == 'view' or param == 'compare'
+        if configurable and i + 1 < len(param_list) and (param_list[i + 1]).isdigit():
+            version_number = int(param_list.pop(i + 1))
+            if version_number > len(versions) or version_number < 1:
+                raise WIKI_PAGE_NOT_FOUND_ERROR
+            elif version_number == len(versions):
+                view_settings[param] = True
+            else:
+                view_settings[param] = version_number
+        else:
+            view_settings[param] = True
+
+    # Build styles for default panel display
+    panels = ['view', 'edit', 'compare']
```

Comment:
Yes and no. Because this logic is used to define which columns are visible on page load and their widths, I'm not sure there's a way to do this on the client without flashing the columns on page load.